### ISSUE TITLE
HTTP 400 from REST API when CosmosDB SqlContainers have TAB in name

### ISSUE THREAD

**Issue by mikeclayton** [Azure/azure-sdk-for-net#14580]
**Describe the bug**

This is possibly more an issue with the server-side REST API than the client library, but I'm not sure where to report it, so let me know if it should go somewhere else...

In short, I've inherited responsibility for a CosmosDB SQL Container which has *somehow* got a tab character in its name, and when I try to use the CosmosDb Client to query this container I get HTTP 400 errors from the API.

I've confirmed the SQL Container's name has a tab in by:

* inspecting it in the results from a call to ```cosmosdbClient.SqlResources.ListSqlContainers``` in Visual Studio - the ```Name``` property on the container shows as ```mySqlContainer\t``` in the debug visualiser

* exporting an ARM template for the CosmosDB Account from the Azure portal, which includes the fragment as follows:

``` 
{
    "type": "Microsoft.DocumentDB/databaseAccounts/sqlDatabases/containers",
    "apiVersion": "2020-06-01-preview",
    "name": "[concat(parameters('databaseAccounts_myDatabaseAccount_name'), '/mySqlDatabase/mySqlContainer\t')]",
   ...
```

The issue appears when trying to use, for example ``` cosmosdbClient.SqlResources.GetSqlContainer( ... )``` or ```cosmosdbClient.SqlResources.GetSqlContainerThroughput( ... )``` for this container - the ```CosmosDBManagementClient``` throws a ```Microsoft.Rest.Azure.CloudException``` with the message ```Operation returned an invalid status code 'BadRequest'```

Capturing the requests in Fiddler shows the following request. Note the ```%09``` encoded tab character in the ```/container/mySqlContainer%09/``` name part of the url - it's technically correct for the name of my container, but possibly unexpected from the REST API's point of view:

```
GET https://management.azure.com/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/myResourceGroup/providers/Microsoft.DocumentDB/databaseAccounts/myDatabaseAccount/sqlDatabases/mySqlDatabase/containers/mySqlContainer%09/throughputSettings/default?api-version=2020-04-01 HTTP/1.1
Host: management.azure.com
x-ms-client-request-id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
Accept-Language: en-US
Authorization: Bearer xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
User-Agent: FxVersion/4.700.20.26901 OSName/Windows OSVersion/Microsoft.Windows.10.0.19041 Microsoft.Azure.Management.CosmosDB.CosmosDBManagementClient/1.2.0.0
```
with a response
```
HTTP/1.1 400 Bad Request
Content-Type: text/html; charset=us-ascii
Server: Microsoft-HTTPAPI/2.0
Date: Wed, 26 Aug 2020 11:08:36 GMT
Connection: close
Content-Length: 324

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN""http://www.w3.org/TR/html4/strict.dtd">
<HTML><HEAD><TITLE>Bad Request</TITLE>
<META HTTP-EQUIV="Content-Type" Content="text/html; charset=us-ascii"></HEAD>
<BODY><h2>Bad Request - Invalid URL</h2>
<hr><p>HTTP Error 400. The request URL is invalid.</p>
</BODY></HTML>
```

**Expected behavior**
Maybe the tab character is supposed to be illegal in CosmosDB SQL Container names, so the actual behaviour for the API is undefined, and the error is expected?

However, given that I *have* got a SQL Container with a tab in its name, I was hoping I would get a HTTP 200 with the relevant details from the API endpoint.

**Actual behavior (include Exception or Stack Trace)**
The CosmosDB client throws a CloudException due to a HTTP 400 from the REST API.

**To Reproduce**

1. Create a CosmosDB SQL Container that has a tab in it (I've not tried to do this, so maybe it's not normally allowed and my container is just an aberration!)

2. Use the ```CosmosDBManagementClient``` to call ```cosmosdbClient.SqlResources.GetSqlContainer``` or ```cosmosdbClient.SqlResources.GetSqlContainerThroughput``` for the container with a tab in its name.

3. The client throws a ```CloudException``` with the message ```Operation returned an invalid status code 'BadRequest'```

Alternatively, this PowerShell script will reproduce the issue with the REST API:

```
$bearer = "myBearerToken";
$subscriptionId = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx";
$resourceGroupName = "myResourceGroupName";
$databaseAccount   = "myDatabaseAccountName";
$sqlDatabaseName   = "mySqlDatabaseName";
$sqlContainerName  = "mySqlContainerName`t"; # where the container name has a trailing tab

$containerUri = "https://management.azure.com" +
                    "/subscriptions/$subscriptionId" +
                    "/resourceGroups/$resourceGroupName" +
                    "/providers/Microsoft.DocumentDB" +
                    "/databaseAccounts/$databaseAccount" +
                    "/sqlDatabases/$sqlDatabaseName" +
                    "/containers/$sqlContainerName" +
                    "?api-version=2020-06-01-preview";

$response = Invoke-RestMethod -Uri $containerUri `
                              -Method "GET" `
                              -Headers @{
                                  "Authorization" = "Bearer $bearer"
                              } `
                              -UseBasicParsing;

# result
#Invoke-RestMethod :
# Bad Request
# Bad Request - Invalid URL
# HTTP Error 400. The request URL is invalid.
# At line:1 char:13
# + $response = Invoke-RestMethod -Uri $throughputUri `
# +             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#     + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], # WebException
#     + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
```

**Environment:**
 - Name and version of the Library package used: [e.g. Azure.Storage.Blobs 12.2.0] 
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): [e.g. Azure AppService or Windows 10 .NET Framework 4.8]
 - IDE and version : [e.g. Visual Studio 16.3]

**Comment by mikeclayton** [comment]
Hi @MehaKaushik - we ended up deleting the container with a tab in its name and recreating it without any weird characters in the name, so the API calls work find now.

I think the root issue was that we'd somehow managed to create a Cosmos DB SQL Container with a trailing tab character in the name - e.g. "myContainer\t". I've tried to create a new one with a tab character in it, but it fails via the REST API, from ```Get-AzCosmosDBSqlContainer``` and in an ARM template, so I don't know how we managed to do that in the first place.

In any case, our immediate problem is solved now.
